[DOCS] Profile Has Privileges API (#87360)
elastic/elasticsearch: x-pack/docs/en/rest-api/security/has-privileges-user-profile.asciidoc

[role="xpack"]
[[security-api-has-privileges-user-profile]]
=== Has privileges user profile API
++++
<titleabbrev>Has privileges user profile</titleabbrev>
++++
[[security-api-has-privileges-user-profile]]

beta::[]

Determines whether the users with the specified <<user-profile, user profile>> IDs
have all the requested privileges.

[[security-api-has-privileges-user-profile-request]]
==== {api-request-title}

`GET /_security/profile/_has_privileges`

`POST /_security/profile/_has_privileges`

[[security-api-has-privileges-user-profile-prereqs]]
==== {api-prereq-title}

To use this API, you must have the `manage_user_profile` cluster privilege.

[[security-api-has-privileges-user-profile-desc]]
==== {api-description-title}

This API is similar to the <<security-api-has-privileges>> API, which checks the privileges only of the user
that's calling it.
However, this API uses the profile IDs, as returned by <<security-api-activate-user-profile>>,

> to identify the users for which to check the privileges of.

Author: Good suggestion! I have pushed [445bf23](https://github.com/elastic/elasticsearch/pull/87360/commits/445bf23874852defa2010e89f264aac45a5f8658) .

them regardless of the value of `allow_restricted_indices`.
`privileges`:::: (list) A list of the privileges that you want to check for the
specified indices.
`application`:::
`application`:::: (string) The name of the application.
`privileges`:::: (list) A list of the privileges that you want to check for the
specified resources. May be either application privilege names, or the names of
actions that are granted by those privileges.
`resources`:::: (list) A list of resource names against which the privileges
should be checked.

Note that the `privileges` section above is identical to the
<<security-api-has-privileges-request-body, request body of the other Has Privileges API>>.

[[security-api-has-privileges-user-profile-response-body]]
==== {api-response-body-title}

A successful has privileges user profile API call returns a JSON structure that contains
two list fields:

`has_privilege_uids`:: (list) The subset of the requested profile IDs of the users that have
**all** the requested privileges.

`error_uids`:: (list) The subset of the requested profile IDs for which an error was
encountered. It does **not** include the missing profile IDs or the profile IDs of

> the users that do not have all the requested privileges. This field is absent if empty.

Reviewer: This gives examples about what `error_uids` is not. Might be useful to give an example of what it can be.

Author: I think documenting the error cases is hard to maintain and not very useful.
Highligting that this is not the "not_has_privileges" field is enough IMO.

[[security-api-has-privileges-user-profile-example]]
==== {api-examples-title}

> The following example checks whether the two users identified by the given profile IDs have all the

Reviewer: I feel `associated` reads better. Also I think we can drop `IDs` for simplicity given the context is clear. WDYT?
```suggestion
The following example checks whether the two users associated to the given profile have all the
```
